textdb/textdb-api/src/test/java/edu/uci/ics/textdb/api/common/SchemaTest.java

```diff
@@ -43,4 +42,22 @@ public void testGetIndex(){
         Assert.assertEquals(expectedIndex_2, retrievedIndex_2);
         
     }
+    
+    @Test(expected = UnsupportedOperationException.class)
+    public void testAddingNewAttribute(){ //Should fail

```

Comment:
I have some high-level questions:

- Why do want to make it immutable?
- Why don't we make the immutability complete?